src/devices/CharacterLcd/LcdConsole.cs

```diff
@@ -0,0 +1,537 @@
+                {
+                    text = text.Substring(0, Size.Width);
+                }
+                else if (text.Length < Size.Width)
+                {
+                    text = text + new string(' ', Size.Width - text.Length);
+                }
+
+                WriteCurrentLine(text);
+            }
+        }
+
+        /// <summary>
+        /// Find where we need to insert additional newlines
+        /// </summary>
+        private void FindLineWraps(int cursorPos, List<string> lines)
+        {
+            if (LineFeedMode.HasFlag(LineFeedMode.Intelligent) || LineFeedMode.HasFlag(LineFeedMode.UntilFull))
+            {
+                int roomOnLine = Size.Width - cursorPos;
+                roomOnLine = Math.Max(roomOnLine, 0);
+                for (int i = 0; i < lines.Count; i++)
+                {
+                    string remaining = lines[i];
+                    if (remaining.Length > roomOnLine)
+                    {
+                        if (LineFeedMode.HasFlag(LineFeedMode.Intelligent))
+                        {
+                            // In intelligent mode, try finding spaces backwards
+                            // Note: this indexes the element 1 char after the last to be printed in the first iteration, since there might be a space just there
+                            int tempRoomOnLine = roomOnLine;
+                            while (remaining[tempRoomOnLine] != ' ' && tempRoomOnLine > 0)
+                            {
+                                tempRoomOnLine--;
+                            }
+
+                            // If we didn't find a space, break after the maximum number of chars
+                            if (tempRoomOnLine > 0)
+                            {
+                                roomOnLine = tempRoomOnLine;
+                            }
+                        }
+
+                        // This line does not fit on the remaining room of this row
+                        string firstPart = remaining.Substring(0, roomOnLine);
+
+                        lines[i] = firstPart;
+                        // Insert the remaining part as new entry to the list, continue iteration there
+                        string remainingPart = remaining.Substring(roomOnLine).TrimStart(' ');
+                        lines.Insert(i + 1, remainingPart);
+
+                    }
+                    roomOnLine = Size.Width; // From now on, we have the full width available
+                }
+            }
+        }
+
+        /// <summary>
+        /// Writes a newline to the display (wrapping to the next line)
+        /// </summary>
+        public void WriteLine()
+        {
+            NewLine();
+        }
+
+        /// <summary>
+        /// Writes the given text to the current position, then wraps to the next line
+        /// </summary>
+        /// <param name="text">Text to draw.</param>
+        public void WriteLine(string text)
```

Comment:
would it make sense to remove the no parmater overload and make this one default to empty? (you can check for string.IsNullEmpty in Write)

Reply:
Not sure. I once heard that overloads are preferrable over default arguments (I think that is/was even an FxCop rule), but I'm not sure that still holds true today.

Reply:
I don't think anyone follows that, I'd prefer we kept just 1 method until someone complains during API review in the future

Reply:
Ok, done.